apps/dotcom/sync-worker/src/TLDrawDurableObject.ts

```diff
@@ -518,17 +519,10 @@ export class TLDrawDurableObject extends DurableObject {
 			if (this.documentInfo.appMode === 'duplicate') {
 				assert(this.documentInfo.duplicateId, 'duplicateId must be present')
 				// load the duplicate id
-				let data: string | undefined = undefined
-				try {
-					const otherRoom = this.env.TLDR_DOC.get(
-						this.env.TLDR_DOC.idFromName(`/${ROOM_PREFIX}/${this.documentInfo.duplicateId}`)
-					) as any as TLDrawDurableObject
-					data = await otherRoom.getCurrentSerializedSnapshot()
-				} catch (_e) {
-					data = await this.r2.rooms
-						.get(getR2KeyForRoom({ slug: this.documentInfo.duplicateId, isApp: true }))
-						.then((r) => r?.text())
-				}
+				await getRoomDurableObject(this.env, this.documentInfo.duplicateId).awaitPersist()
```

Comment:
I guess this still does not guarantee the data to be up to date since we are using an `ExecutionQueue`? Push to it will happen, but the task might not run in time?

Reply:
it waits for the enqueued task to complete